### PR TITLE
CLC-5456 - Unscheduled task edits respond differently to mouse click vs keystroke

### PR DESCRIPTION
**Pull request by christianv**
https://jira.ets.berkeley.edu/jira/browse/CLC-5456

### CODE REVIEW

File: src/assets/javascripts/angular/services/widgetService.js [hunk 1]
```diff
@@ -6,8 +6,11 @@
      * Toggle whether an item for a widget should be shown or not
      */
     var toggleShow = function(event, items, item, widget) {
+      var tagName = (event && event.toElement && event.toElement.tagName);
       // Ignore toggling on Anchor events
-      if (event && event.toElement && event.toElement.tagName === 'A') {
+      if (tagName === 'A' ||
+          tagName === 'INPUT' ||
+          tagName === 'TEXTAREA') {
```
Comment (redconfetti) [inline src/assets/javascripts/angular/services/widgetService.js]: Would this be shorter using something like:
```javascript
if (['A', 'INPUT', 'TEXTAREA'].indexOf(tagName)) { // ... }
```

Comment (christianv) [inline src/assets/javascripts/angular/services/widgetService.js]: @redconfetti yes, way better, updated the PR

File: src/assets/javascripts/angular/services/widgetService.js [hunk 2]
```diff
@@ -6,8 +6,9 @@
      * Toggle whether an item for a widget should be shown or not
      */
     var toggleShow = function(event, items, item, widget) {
+      var tagName = (event && event.toElement && event.toElement.tagName);
       // Ignore toggling on Anchor events
-      if (event && event.toElement && event.toElement.tagName === 'A') {
+      if (['A', 'INPUT', 'TEXTAREA'].indexOf(tagName) !== -1) {
```
Comment (redconfetti) [inline src/assets/javascripts/angular/services/widgetService.js]: @christianv Oh, I take it that when it returns 0, that's considered false?

Comment (christianv) [inline src/assets/javascripts/angular/services/widgetService.js]: @redconfetti indeed, `indexOf` will return the position of the element in the array (starting with 0).
A quick JSBin to show the result: http://jsbin.com/sovotedaye/edit?js,console

**Pull request merged**